nautobot/extras/api/views.py

```diff
@@ -441,9 +441,18 @@ def sync(self, request, pk):
             raise CeleryWorkerNotRunningException()
 
         repository = get_object_or_404(GitRepository, id=pk)
-        repository.sync(user=request.user)
-        return Response({"message": f"Repository {repository} sync job added to queue."})
 
+        job_result = repository.sync(user=request.user)
+
+        return Response({
+                "message": f"Repository {repository} sync job added to queue.",
+                "repository_id": repository.id,
+                "repository_name": repository.name,
+                "sync_jobmodel_id": job_result.job_model_id,
+                "sync_jobresult_id": job_result.id,
```

Comment:
It'd probably be better to have a `job_result` key and get its value as a dict from a `serializers.JobResultSerializer(job_result, context={"request": request}).data` call - similar to `JobViewSetBase.run()`.

Reply:
Noted; I'll do this. I absolutely want to do it the "right way".

Reply:
Done, I think!

Reply:
Done; I hope I've understood correctly. See what you think.